### DISCUSSION TITLE
How to filter listings by multiple categories?

### DISCUSSION THREAD

**Discussion by Adithya1243** [quarto-dev/quarto-cli#4270] [Q&A]
Hi. 

I have created a grid listing and noticed that my content are filtered using only 1 category at a time. While this is great, I couldn't find much documentation on how to filter content using multiple categories. 

For example, there are 5 posts each with the following categories ( git, docker, tensorflow, rstudio, sql). Usually I would be able to filter posts by selecting any one of them. But I want to select 2 categories at once, so that I would see 2 posts. 

Can anyone please help?

**Comment by gdurif** [comment]
Hello,
With `filter_ui: true` in the listing options, the filtering also search in the categories, so typing "git docker" will behave as if you were selecting both categories.
That is a bit of a workaround but it works.
Best

**Comment by christina-kaiser** [comment]
Hi,

did you find a solution? While the one gdurif provided works, it is not very intuitive. It would be much better to be able to select two categories as you wrote... 

Best regards

**Comment by Adithya1243** [comment]
Hi,
I didn't find anything useful in Quarto. I had to write my custom CSS,
Javascript code for the multi select functionality.

**Reply by christina-kaiser** [reply]
cool, would you be willing to share the code? :-)

**Comment by haxfn** [comment]
Should I create a new discussion about this? @mcanouil ?

**Reply by haxfn** [reply]
![image](https://github.com/user-attachments/assets/27ebf46e-b46f-42b4-80c6-52c2ba8bc4bc)

something like this is important for my use case.

**Reply by mcanouil** [reply]
No, as the only solution is to write your own piece of code to do that.

**Comment by andrewpbray** [comment]
I also ran into the need for the feature. @Adithya1243 if you'd be willing to share your code, that'd be great, otherwise I can work on writing this an packaging it into an extension.

**Reply by mcanouil** [reply]
I actually have a small modification of the original quarto listing javascript for this.
I'll submit a PR to the team soon, once I have something that does not break the "all" selection part.

https://github.com/user-attachments/assets/9dd8e467-f4a3-4c34-98a9-8ee9a4b8268e

**Reply by mcanouil** [reply]
Here is the draft:

- https://github.com/quarto-dev/quarto-cli/pull/12667

The easiest way to try it out is to take `quarto-listing.js` and use post-render script to replace the one from Quarto in your project, *i.e.*, `_site/site_libs/quarto-listing/quarto-listing.js`.

**Reply by andrewpbray** [reply]
Hey this looks great!
